pkg/cmd/samples.go

```diff
@@ -1,88 +1,27 @@
 package cmd
 
 import (
-	"fmt"
-	"os"
-
-	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
-	"github.com/stripe/stripe-cli/pkg/ansi"
-	gitpkg "github.com/stripe/stripe-cli/pkg/git"
-	"github.com/stripe/stripe-cli/pkg/samples"
-	"gopkg.in/src-d/go-git.v4"
+
+	"github.com/stripe/stripe-cli/pkg/cmd/samples"
 )
 
 type samplesCmd struct {
-	cmd *cobra.Command
+	Cmd *cobra.Command
 }
 
-func newSamplesCmd() *samplesCmd {
-	return &samplesCmd{
-		cmd: &cobra.Command{
-			// TODO: create subcommand
-			// TODO: list subcommand
+func newSamplesCmd() *cobra.Command {
+	samplesCmd := &samplesCmd{
+		Cmd: &cobra.Command{
 			// TODO: fixtures subcommand
-			Use: "samples",
-			RunE: func(cmd *cobra.Command, args []string) error {
-				sample := samples.Samples{
-					Config: Config,
-					Fs:     afero.NewOsFs(),
-					Git:    gitpkg.Operations{},
-				}
-				selectedSample := args[0]
-
-				spinner := ansi.StartSpinner(fmt.Sprintf("Downloading %s", selectedSample), os.Stdout)
-				// Initialize the selected sample in the local cache directory.
-				// This will either clone or update the specified sample,
-				// depending on whether or not it's. Additionally, this
-				// identifies if the sample has multiple integrations and what
-				// languages it supports.
-				err := sample.Initialize(selectedSample)
-				if err != nil {
-					switch e := err.Error(); e {
-					case git.NoErrAlreadyUpToDate.Error():
-						// Repo is already up to date. This isn't a program
-						// error to continue as normal
-						break
-					case git.ErrRepositoryAlreadyExists.Error():
-						// If the repository already exists and we don't pull
-						// for some reason, that's fine as we can use the existing
-						// repository
-						break
-					default:
-						ansi.StopSpinner(spinner, "An error occurred.", os.Stdout)
-						return err
-					}
-				}
-				ansi.StopSpinner(spinner, "Finished downloading.", os.Stdout)
-
-				// Once we've initialized the sample in the local cache
-				// directory, the user needs to select which integration they
-				// want to work with (if selectedSamplelicable) and which language they
-				// want to copy
-				err = sample.SelectOptions()
-				if err != nil {
-					return err
-				}
-
-				// Create the target folder to copy the sample in to. We do
-				/// this here in case any of the steps above fail, minimizing
-				// the change that we create a dangling empty folder
-				targetPath, err := sample.MakeFolder(selectedSample)
-				if err != nil {
-					return err
-				}
-
-				// Perform the copy of the sample given the selected options
-				// from the selections above
-				err = sample.Copy(targetPath)
-				if err != nil {
-					return err
-				}
-
-				// TODO: setup .env
-				return nil
-			},
+			Use:   "samples",
+			Short: `Sample integrations built by Stripe`,
+			Long:  ``,
 		},
 	}
+
+	samplesCmd.Cmd.AddCommand(samples.NewCreateCmd(&Config))
+	samplesCmd.Cmd.AddCommand(samples.NewListCmd())
+
+	return samplesCmd.Cmd
```

Comment:
Any reason you're deviating from the usual pattern of returning the `whateverCmd` struct and then calling `rootCmd.AddCommand(whateverCmd.cmd)`?

Reply:
Noticed this as well! It looks a little deviant in the list in `root.go` above.

Reply:
I changed this because we don't actually use the returned cmd struct anywhere so it seemed unnecessary to export it, but at that point the linter is unhappy because we're returning an un-exported type. I'll change this back for now.